config_ipa.go

```diff
@@ -46,41 +46,34 @@ const (
 	EmptyCodeHashSecondHalfIdx = EmptyCodeHashFirstHalfIdx + 1
 )
 
-type IPAConfig struct {
-	conf *ipa.IPAConfig
-}
+var (
+	FrZero Fr
+	FrOne  Fr
 
-type Config = IPAConfig
+	cfg     *Config
+	onceCfg sync.Once
+)
 
-func (ipac *IPAConfig) CommitToPoly(poly []Fr, _ int) *Point {
-	ret := ipac.conf.Commit(poly)
-	return &ret
+func init() {
+	FrZero.SetZero()
+	FrOne.SetOne()
 }
 
-var cfg *Config
+type IPAConfig struct {
+	conf *ipa.IPAConfig
+}
 
-var precompFileName = "precomp"
+type Config = IPAConfig
 
 func GetConfig() *Config {
-	if cfg == nil {
-		var ipacfg *ipa.IPAConfig
-		if precompSer, err := os.ReadFile(precompFileName); err != nil {
-			ipacfg = ipa.NewIPASettings()
-			serialized, err := ipacfg.SRSPrecompPoints.SerializeSRSPrecomp()
-			if err != nil {
-				panic("error writing serialized precomputed Lagrange points:" + err.Error())
-			} else if err = os.WriteFile(precompFileName, serialized, 0666); err != nil {
-				panic("error saving the precomp: " + err.Error())
-			}
-		} else {
-			srs, err := ipa.DeserializeSRSPrecomp(precompSer)
-			if err != nil {
-				panic("error deserializing precomputed Lagrange points:" + err.Error())
-			}
-			ipacfg = ipa.NewIPASettingsWithSRSPrecomp(srs)
+	onceCfg.Do(func() {
+		conf, err := ipa.NewIPASettings()
+		if err != nil {
+			panic(err)
```

Comment:
The previous code panicked, and I was very tempted to allow `GetConfig()` to return an error.
It should be impossible to have an error in the way we call it here. The only way `NewIPASettings()` an error is that if you provide less than 256-points (internally), which is "never".

If we return an `error` in `GetConfig()`, that can be a bit annoying in calling sites that don't return an error will have to do it or also decide to panic.

For now leaving this as a comment since this isn't adding a new panic case. If this panics, it'd be at node startup time.